backend/typescript/rest/givecloud.ts

```diff
@@ -0,0 +1,56 @@
+import { Router } from "express";
+import AuthService from "../services/implementations/authService";
+import UserService from "../services/implementations/userService";
+import { sendErrorResponse } from "../utilities/errorResponse";
+import { AuthDTO, Role, UserDTO } from "../types";
+
+const givecloudRouter: Router = Router();
+
+const authDtoToToUserDto = (authDTO: AuthDTO): UserDTO => {
+  const userDTO: UserDTO = (({
+    id,
+    firstName,
+    lastName,
+    email,
+    roleType,
+    subscriptionExpiresOn,
+  }) => ({ id, firstName, lastName, email, roleType, subscriptionExpiresOn }))(
+    authDTO,
+  );
+  return userDTO;
+};
+
+const userService = new UserService();
+const authService = new AuthService(userService);
+
+givecloudRouter.post("/user.subscription_paid", async (req, res) => {
```

Comment:
Could you close this route to the public (ie add the `isAuthenticatedByRole("Admin")` middleware) temporarily while we implement the secret verification? Leaving this open may be a security vulnerability